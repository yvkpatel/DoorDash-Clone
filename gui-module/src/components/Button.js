const Button = ({text, color, onClick}) => {
    return(
        <button className='button'
        style={{background:color}}
        data-testid="button"
        onClick={onClick}>
            {text}
        </button>
    )
}

export default Button