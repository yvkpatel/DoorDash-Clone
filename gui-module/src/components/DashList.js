/*  DashList.js
    Component that acts as a container for cards displayed in a veritcal list.

    === Props ===
    itemList = list of the items to be displayed in each of the cards.
        Example: const [listVar] = useState([{ id: 1, title: 'Card 1'}, { id: 2, title: 'Card 2'}]);

    component = the component that will be used for each of the cards.

    manager = class that will provide methods that may be called from the card.
*/

const List = ({ itemList, component, manager }) => {
  const Component = component;

  return (
    <div className="list" data-testid="dash-list">
      {itemList
        ? itemList.map((item) => (
            <Component key={item.id} item={item} manager={manager} />
          ))
        : "Loading:..."}
    </div>
  );
};

export default List;
