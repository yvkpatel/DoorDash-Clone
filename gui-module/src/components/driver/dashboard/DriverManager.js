/*  DriverManager.js
    Provides methods for the driver dashboard to interact with databases and manage the driver's local information.
*/
class DriverManager {
  constructor(id, income, orders) {
    this.id = id;
    this.income = income;
    this.orders = orders;
  }

  // =============== LOCAL MANAGEMENT METHODS ===============

  // Converts the driver's income from a float to a currency formatted string.
  getIncome() {
    return this.income + "$";
  }

  updateIncome(x) {
    this.income += x;
  }

  // =============== METHODS THAT INTERACT WITH DATABASES ===============

  // PLACEHOLDER: will fetch driver tasks from database.
  // Current implementation simply returns a hardcoded list of imaginary tasks.
  getDriverOrders() {
    // @JR Note: some hefty refactoring might have to take place on DriverDash.js
    // depends on how the database fetches.
    return [
      {
        id: 1,
        title: "John Doe",
        restaurant: "Boston Pizza",
        destination: "1600 Pennsylvania Avenue",
      },
      // {
      //     id: 2,
      //     title: "Famous Pizza, I guess.",
      //     restaurant: "Dominos",
      //     destination: "4059 Mt Lee Dr. Hollywood"
      // },
      // {
      //     id: 3,
      //     title: "Pentagram Pizza",
      //     restaurant: "Pizza Hut",
      //     destination: "6114 California Street"
      // },
      // {
      //     id: 4,
      //     title: "Curious Pizza",
      //     restaurant: "???",
      //     destination: "221B Baker Street"
      // },
      // {
      //     id: 5,
      //     title: "Pizza, dude.",
      //     restaurant: "Literally Anywhere",
      //     destination: "122 + 1/8th Bleaker Street"
      // },
    ];
  }

  // PLACEHOLDER: When called it should inform database that the order is now taken.
  driverAcceptOrder(orderID) {
    console.log("Order ID: " + orderID);
  }
}

export default DriverManager;
