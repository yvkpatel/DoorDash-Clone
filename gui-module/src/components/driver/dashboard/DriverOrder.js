// Displays orders to driver to accept
import React from "react";
import Button from "../../Button";
import { Link } from "react-router-dom";

// TODO: Renaming CSS classes to be more generic.

const DriverOrder = (props) => {
  return (
    <div className="driverTask">
      <div className="driverTaskHeader">
        <h2>{props.item.title}</h2>
        <Link
          to={{ pathname: "../drivermap", state: { orderId: props.item.id } }}
        >
          <Button
            text="Accept"
            color="#2DB6F0"
            onClick={() => props.manager.driverAcceptOrder(props.item.id)}
          />
        </Link>
      </div>
      <div className="driverTaskDetails">
        <p>Pick up: {props.item.restaurant}</p>
        <p>Deliver: {props.item.destination}</p>
      </div>
    </div>
  );
};

export default DriverOrder;
