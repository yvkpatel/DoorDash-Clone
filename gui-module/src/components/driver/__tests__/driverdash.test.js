import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import DriverDash from "../dashboard/DriverDashPage";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import ReactDOM from "react-dom";
import Button from "../../Button";
import DashList from "../../DashList";
import React, { useEffect } from "react";
import DriverOrder from "../dashboard/DriverOrder";
import DriverManager from "../dashboard/DriverManager";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

test("should render driver dashboard component", () => {
  render(
    <BrowserRouter>
      <DriverDash />
    </BrowserRouter>
  );
  const DriverDashElement = screen.getByTestId("driver-dash");
  expect(DriverDashElement).toBeDefined();
});

test("List of restaurants renders", () => {
  render(
    <BrowserRouter>
      <DriverDash />
    </BrowserRouter>
  );
  const DriverDashElement = screen.getByTestId("dash-list");
  expect(DriverDashElement).toBeDefined();
});

test("Accept button renders", () => {
  render(
    <BrowserRouter>
      <DriverDash />
    </BrowserRouter>
  );
  const DriverDashElement = screen.getAllByRole("button", { name: "Accept" });
  expect(DriverDashElement).toBeDefined();
});

const mockRestaurant = {
  item: {
    id: "1",
    title: "Presidential Pizza",
    restaurant: "a",
    destination: "1600 Pennsylvania Avenue",
  },
  manager: {
    id: 4,
    income: 9.24,
    orders: 2,
    driverAcceptOrder(orderID) {
      console.log("Order ID: " + orderID);
    },
  },
};

test("Clicking the Accept button and going to the maps page", () => {
  render(
    <BrowserRouter>
      <DriverOrder {...mockRestaurant} />
    </BrowserRouter>
  );
  fireEvent.click(screen.getByText("Accept"));
  const newURL = window.location.href;
  expect(newURL).toBe("http://localhost/drivermap");
});

test("matches snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <DriverDash />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
