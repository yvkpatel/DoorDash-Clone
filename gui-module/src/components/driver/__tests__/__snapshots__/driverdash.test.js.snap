// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
    <a
      className="accSettings"
      href="https://developer.mozilla.org/en-US/docs/Web/CSS"
    >
      Account Settings
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "#2DB6F0",
          }
        }
      >
        Logout
      </button>
    </a>
  </header>
  <div
    className="driverDashMain"
  >
    <div
      className="dashInfoBox"
      data-testid="driver-dash"
    >
      <h1>
        Account Summary
      </h1>
      <p>
        Income: 
        9.24$
      </p>
      <p>
        Orders: 
        2
      </p>
    </div>
    <div
      className="list"
      data-testid="dash-list"
    >
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            John Doe
          </h2>
          <a
            href="../drivermap"
            onClick={[Function]}
          >
            <button
              className="button"
              data-testid="button"
              onClick={[Function]}
              style={
                Object {
                  "background": "#2DB6F0",
                }
              }
            >
              Accept
            </button>
          </a>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Pick up: 
            Boston Pizza
          </p>
          <p>
            Deliver: 
            1600 Pennsylvania Avenue
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
